server/task_executor.go

```diff
@@ -27,7 +27,16 @@ type TaskExecutor interface {
 	RunOnce(ctx context.Context, server *Server, task *api.Task) (terminated bool, result *api.TaskRunResultPayload, err error)
 }
 
-func runMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.Task, migrationType db.MigrationType, statement, schemaVersion string, vcsPushEvent *vcsPlugin.PushEvent) ( /*terminated*/ bool /*result*/, *api.TaskRunResultPayload, error) {
+type migrationExecutorContext struct {
+	databaseName  string
+	repoRawOutter *api.RepositoryRaw
+	mi            *db.MigrationInfo
+	issue         *api.Issue
+	migrationID   int64
+	schema        string
+}
+
+func (mc *migrationExecutorContext) preMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.Task, migrationType db.MigrationType, statement, schemaVersion string, vcsPushEvent *vcsPlugin.PushEvent) (bool, *api.TaskRunResultPayload, error) {
```

Comment:
It's a bit strange to have a context (data object) to have action functions from OO design point of view. Shall we just change preMigration() to PreMigration()?